src/python/pants/core/goals/lint.py

```diff
@@ -209,11 +221,13 @@ class SubPartition(Generic[_PartitionElementT]):
 
     @final
     @classmethod
-    def registration_rules(cls) -> Iterable[UnionRule]:
-        yield from cls._get_registration_rules()
+    def registration_rules(
+        cls, *, partitioner_type: PartitionerType = PartitionerType.CUSTOM
```

Comment:
The default implementations (here and in `fmt`) are enough to warrant testing. So blocking my own PR until that's here.